client/echo_client.go

```diff
@@ -188,15 +215,18 @@ func NewEchoClient(ctx context.Context, opts ...option.ClientOption) (*EchoClien
 	if err != nil {
 		return nil, err
 	}
-	c := &EchoClient{
+	client := EchoClient{CallOptions: defaultEchoCallOptions()}
+
+	c := &echoGrpcClient{
 		connPool:         connPool,
 		disableDeadlines: disableDeadlines,
-		CallOptions:      defaultEchoCallOptions(),
-
-		echoClient: genprotopb.NewEchoClient(connPool),
+		echoClient:       genprotopb.NewEchoClient(connPool),
+		CallOptions:      &client.CallOptions,
 	}
 	c.setGoogleClientInfo()
 
+	client.internalEchoClient = c
+
 	c.LROClient, err = lroauto.NewOperationsClient(ctx, gtransport.WithConnPool(connPool))
```

Comment:
I think the same thing that happened with the CallOptions needs to happen to the `LROClient` - the wrapper client should get the initially created one, and the internal client should get a reference to it.